### PR TITLE
Consolidate Data structure

### PR DESCRIPTION
**Pull request by taysea**
This PR consolidates the page data into a single structure.js file. Previously data was being held in various places, causing some repeats in data. The goal of this is to consolidate the data in one place.

Additionally, uses of the term "prefix" were replaced with "topic".

### CODE REVIEW

File: aries-site/src/data/structure.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  IconResources,
+  IconBrand,
+  IconColor,
+  IconIcons,
+  IconLayout,
+  IconType,
+  IconToken,
+} from '../components/icons';
+
+export const structure = [
```
Comment (halocline) [inline aries-site/src/data/structure.js]: Now that this is flattened, `pages` might be more precise that `structure`. Not a big deal either way.

File: aries-site/src/pages/index.js [hunk 2]
```diff
@@ -29,7 +30,7 @@ const Index = () => {
     }
   }, []);
 
-  const data = Object.values(homeData);
+  const data = structure.filter(page => page.pages);
```
Comment (halocline) [inline aries-site/src/pages/index.js]: `data` is a bit generic here, wonder if `topicList` is more explicit and would communicate to developer what this data represents?

Comment (halocline) [inline aries-site/src/pages/index.js]: What happens if a content page gets populated with a `pages` property?  Won't it now errantly get picked up in this filter?

What if we gave the object in the data structure a property of `parent: 'Home'` and the modified the filter to be `const data = structure.filter(page => page.parent === title` (Where title is a variable that then populates the `<PageLayout>` title as well)?   This way, we will only receive pages for which we are explicitly asking.

File: aries-site/src/layouts/main/SidebarLayout.js [hunk 3]
```diff
@@ -27,8 +28,8 @@ export const SidebarLayout = ({ mainContentChildren }) => {
 
   // TODO selectedNav should be retrived from aries-core
   // as the selected element of the NavBar
-  const selectedNav = 'foundation';
-
+  const topic = 'Foundation';
```
Comment (halocline) [inline aries-site/src/layouts/main/SidebarLayout.js]: We are going to have to rework how this topic variable gets populated. It should get passed in as a prop from the component consuming `<SidebarLayout>` --- outside the scope of this PR though.

Comment (taysea) [reply aries-site/src/layouts/main/SidebarLayout.js]: Agreed.

Comment (taysea) [reply aries-site/src/layouts/main/SidebarLayout.js]: Agreed.

File: aries-site/src/data/structure.js (hunk 1 above)
Comment (taysea) [reply aries-site/src/data/structure.js]: I like pages. I'll make a switch.

File: aries-site/src/pages/index.js (hunk 2 above)
Comment (taysea) [reply aries-site/src/pages/index.js]: I like this. I'll add that in.

Comment (taysea) [reply aries-site/src/pages/index.js]: Agreed, we can be more specific.

File: aries-site/src/pages/resources/index.js [hunk 4]
```diff
@@ -1,26 +1,24 @@
 import React from 'react';
-import { PageLayout, NavPage, SideBarItemList } from '../../layouts';
+import { PageLayout, NavPage } from '../../layouts';
 import { DescriptiveHeader } from '../../components/headings';
-import { data } from '../../components/home';
+import { structure } from '../../data';
 
 const title = 'Resources';
-const prefix = title.toLowerCase();
+const [topic] = structure.filter(page => page.name === title);
```
Comment (halocline) [inline aries-site/src/pages/resources/index.js]: good

File: aries-site/src/data/structure.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  {
```
Comment (halocline) [inline aries-site/src/data/structure.js]: To be comprehensive, we should add an object for the Home page.
```
{
  name: 'Home',
  pages: ['Guidelines', ... ],
}
```

Would require modification to this filter mentioned here: https://github.com/hpe-design/aries/pull/168/files?file-filters%5B%5D=.js&hide-deleted-files=true#r363999070

File: aries-site/src/data/structure.js [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    name: 'Guidelines',
+    color: 'green',
+    description:
+      'Why does HPE have a design system? All the aesthetics, best practices, and information about the platform and how to wield it.',
+    icon: size => <IconGuidelines size={size} />,
+    pages: ['Principles', 'Human Centered', 'Philosophy'],
```
Comment (halocline) [inline aries-site/src/data/structure.js]: Recommend adding `parent: 'Home` per this filter comment: https://github.com/hpe-design/aries/pull/168/files?file-filters%5B%5D=.js&hide-deleted-files=true#r363999070

File: aries-site/src/data/structure.js [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+  },
+  {
+    name: 'Foundation',
+    color: 'red',
```
Comment (halocline) [inline aries-site/src/data/structure.js]: This is going to be nice to have.

File: aries-site/src/pages/components/index.js [hunk 8]
```diff
@@ -1,26 +1,24 @@
 import React from 'react';
-import { PageLayout, NavPage, SideBarItemList } from '../../layouts';
+import { PageLayout, NavPage } from '../../layouts';
 import { DescriptiveHeader } from '../../components/headings';
-import { data } from '../../components/home';
+import { structure } from '../../data';
 
 const title = 'Components';
-const prefix = title.toLowerCase();
+const [topic] = structure.filter(page => page.name === title);
```
Comment (taysea) [inline aries-site/src/pages/components/index.js]: Wondering if here we can use the .find() approach as well? @halocline

**Pull request merged**